server/src/main/java/io/druid/segment/realtime/RealtimeManager.java

```diff
@@ -211,79 +200,50 @@ public FireDepartmentMetrics getMetrics(String datasource)
            );
   }
 
-  static class FireChief extends Thread implements Closeable
+  static class FireChief implements Runnable
   {
     private final FireDepartment fireDepartment;
     private final FireDepartmentMetrics metrics;
     private final RealtimeTuningConfig config;
     private final QueryRunnerFactoryConglomerate conglomerate;
 
-    private volatile Firehose firehose = null;
-    private volatile FirehoseV2 firehoseV2 = null;
-    private volatile Plumber plumber = null;
-    private volatile boolean normalExit = true;
+    private Plumber plumber;
 
-    public FireChief(FireDepartment fireDepartment, QueryRunnerFactoryConglomerate conglomerate)
+    FireChief(FireDepartment fireDepartment, QueryRunnerFactoryConglomerate conglomerate)
     {
       this.fireDepartment = fireDepartment;
       this.conglomerate = conglomerate;
       this.config = fireDepartment.getTuningConfig();
       this.metrics = fireDepartment.getMetrics();
     }
 
-    public Firehose initFirehose()
+    private Firehose initFirehose()
     {
-      synchronized (this) {
-        if (firehose == null) {
-          try {
-            log.info("Calling the FireDepartment and getting a Firehose.");
-            firehose = fireDepartment.connect();
-            log.info("Firehose acquired!");
-          }
-          catch (IOException e) {
-            throw Throwables.propagate(e);
-          }
-        } else {
-          log.warn("Firehose already connected, skipping initFirehose().");
-        }
-
-        return firehose;
+      try {
+        log.info("Calling the FireDepartment and getting a Firehose.");
+        return fireDepartment.connect();
+      }
+      catch (IOException e) {
+        throw Throwables.propagate(e);
       }
     }
 
-    public FirehoseV2 initFirehoseV2(Object metaData)
+    private FirehoseV2 initFirehoseV2(Object metaData)
     {
-      synchronized (this) {
-        if (firehoseV2 == null) {
-          try {
-            log.info("Calling the FireDepartment and getting a FirehoseV2.");
-            firehoseV2 = fireDepartment.connect(metaData);
-            log.info("FirehoseV2 acquired!");
-          }
-          catch (IOException e) {
-            throw Throwables.propagate(e);
-          }
-        } else {
-          log.warn("FirehoseV2 already connected, skipping initFirehoseV2().");
-        }
-
-        return firehoseV2;
+      try {
+        log.info("Calling the FireDepartment and getting a FirehoseV2.");
+        return fireDepartment.connect(metaData);
+      }
+      catch (IOException e) {
+        throw Throwables.propagate(e);
       }
     }
 
-    public Plumber initPlumber()
+    Plumber initPlumber()
```

Comment:
There are subclasses calling this method in RealtimeManagerTest. Changed the return type to void.